javascript/javascript2/week1/main.js

```diff
@@ -0,0 +1,93 @@
+//Find the shortest word
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+const shortestWord = document.getElementById('shortest-word');
+const displayWord = document.createElement('h4')
+displayWord.innerHTML = `Danish Words List : [${danishWords}]`;
+shortestWord.appendChild(displayWord);
+function findShortestWord(danishWords) {
+    let shortest = danishWords.reduce((shortestWord, currentWord) => {
+        return currentWord.length < shortestWord.length ? currentWord : shortestWord;
+    }, danishWords[0]);
+    return shortest;
+}
+const shortestOutput = findShortestWord(danishWords);
+const displayOutput = document.createElement('h4')
+displayOutput.innerHTML = `Shortest word from List is : ${shortestOutput}`;
+shortestWord.appendChild(displayOutput);
+// Danish letters
+const danishString = "Jeg har en blå bil";
+const danishString2 = "Blå grød med røde bær";
+const danishLetter = document.getElementById('danish-letters');
+const danishLetters = document.createElement('h4');
+danishLetters.innerHTML = `Danish letters 1 : "${danishString}"` + `Danish letters 2 : "${danishString2} "`;
+danishLetter.appendChild(danishLetters);
+function findDanishString(danishChar) {
+    let charÅ = (danishChar.match(/å/i) || []).length;
+    let charÆ = (danishChar.match(/æ/i) || []).length;
+    let charØ = (danishChar.match(/ø/i) || []).length;
+    let total = charÅ + charÆ + charØ;
+    const displyChar = {
+        Total: total,
+        æ: charÆ,
+        ø: charØ,
+        å: charÅ
+    }
+    return displyChar;
+}
+console.log(findDanishString(danishString))
+console.log(findDanishString(danishString2))
+const danishChar = JSON.stringify(findDanishString(danishString)) + JSON.stringify(findDanishString(danishString2));
+const displayCharOutput = document.createElement('p')
+displayCharOutput.innerHTML = `Results : ${danishChar}`
+danishLetter.appendChild(displayCharOutput);
```

Comment:
Good work! 

A detail to be aware of, is that the Danish letters can be repeated in the strings, in your variable danishString2, we in fact have 4 Danish letters with å: 1, æ: 1, and ø: 2. How can you update your implementation such that it includes the repeated Danish letters, both for the total count and for the individual letter count? 😊